### ISSUE TITLE
Drag&drop to reorder and re-prioritize todos

### ISSUE THREAD

**Issue by page200** [ransome1/sleek#383]
Please allow drag&drop to reorder todos. They can be reordered within their priority class (affecting the order in which they are listed). Dragging them into a different priority class can assign them to that priority class (and that respective position in the priority class).

**Comment by page200** [comment]
The current sorting method seems to be based on Contexts and Projects.

Since every todo can be assigned any subset of Contexts and Projects (i.e. there is an exponential number 2ᴺ of possibilities for each todo), a sorting method based on Contexts and Projects cannot treat Contexts and Projects in a fair manner and in general cannot create logical orders (for example orders where all tasks from each specific Context are always contiguous).

For displaying todos based on Contexts and Projects, **Filters** can be used.

On the other hand, the **order** should be decided by the user. There should be no huge limitations for the user to choose the order. There should be freedom to reorder todos _within_ each priority letter, similarly to editing the `todo.txt` file in plain text.

**Comment by lindoman** [comment]
I totally agree with this. I would also welcome keyboard shortcuts for the feature.

**Comment by page200** [comment]
Then there can be a very nice way to add a new todo:

- Hover the mouse cursor roughly on the horizontal line between two rows of the todo list.
- The horizonal line becomes thicker and gets a different color, indicating that you can click on it.
- Click on it.
- The "new todo" dialog appears.

**Comment by page200** [comment]
Actually, the "new todo" or "edit todo" dialog, instead of being a dialog in the middle of the screen, can be in-place in between two existing todos.
Click on the line between two todos, and that line turns into a "new todo" form (all todos below it slide down).
Click on an existing todo, and it turns into an "edit todo" form (same form as the "new todo" form, but filled in with the properties of the todo you clicked).

**Comment by page200** [comment]
I recently looked through [the issues in the "High Priority" card](https://github.com/ransome1/sleek/projects/2#column-13531090), and the basic form of the present issue (drag&drop to reorder todos) seems more important, in my humble opinion. It would be great to give it a high priority.

Six users gave it a "thumbs up", by the way.

Workarounds are quite cumbersome.

In any case, thank you! :)

**Comment by page200** [comment]
This functionality would be very useful.

**Comment by page200** [comment]
This functionality would be very useful.

**Comment by ucc-et** [comment]
Just as a i guess info. I just implemented the drag and drops (for priorities atleast). 
Thought I'd let ya know, cause I also wanted to have this feature really badly. 
I will try to be as user friendly as possible
